src/mem/alloc.h

```diff
@@ -683,6 +684,11 @@ namespace snmalloc
       if (isFake)
         return;
 
+      // Entropy must be first, so that all data-structures can use the key
+      // it generates.
+      // This must occur before any freelists are constructed.
+      entropy.init<typename MemoryProvider::Pal>();
```

Comment:
We are probably inconsistent about referring to `MemoryProvider::Pal` vs. just the `Pal` that's in scope.  Most often, those are the same, but it might be good to examine all the places we've defaulted `<typename PAL = Pal>`?  Doesn't have to be this PR; maybe just raise an issue so we don't forget?